common/core/gpudevice.cpp

```diff
@@ -518,6 +560,50 @@ void GpuDevice::HandleHWCSettings() {
     temp_displays.swap(total_displays_);
   }
 
+  // Now set floating display configuration
+  // Get the floating display index and the respective rectangle
+  // TODO Logical display on & mosaic display on scenario
+  if (use_float && !use_logical && !use_mosaic) {
+    size_t size = float_display_indices.size();
+    bool ret = false;
+
+    // Set custom resolution to desired display instance
+    for (size_t i = 0; i < size; i++) {
+      std::vector<int32_t> &float_rect = float_displays.at(i);
+      HwcRect<int32_t> rect = HwcRect<int32_t>(float_rect.at(0),
+        float_rect.at(1), float_rect.at(2), float_rect.at(3));
+      ret = total_displays_.at(float_display_indices.at(i))->SetResolution(rect);
+    }
+
+    // Set dummy resolution to other display instances
+    // so that flags are set consistently
+    // TODO remove dummy setting
+    size_t i = 0, j = 0;
+    HwcRect<int32_t> rect_dummy = HwcRect<int32_t> (0,0,0,0);
+    size_t num_displays = total_displays_.size();
+
+    for (i = 0; i < num_displays; i++) {
+      for (j = 0; j < size; j++) {
+        // The display instance already set with custom resolution
+        if (i == float_display_indices.at(j)) {
+	  break;
+	}
+      }
+      if (j == size) {
+        ret = total_displays_.at(i)->SetResolution(rect_dummy);
+      }
+    }
+  } else {
+    // Set dummy resolution to other display instances
+    // so that flags are set consistently
+    // TODO remove dummy setting
+    HwcRect<int32_t> rect_dummy = HwcRect<int32_t>(0,0,0,0);
+    bool ret = false;
+    for (size_t i = 0; i < total_displays_.size(); i++) {
+      ret = total_displays_.at(i)->SetResolution(rect_dummy);
```

Comment:
Do we really need this, if default flag in PhysicalDisplay is false ?

Reply:
We should be able to remove from line 578 till 603. Will do that.